src/udev/net/link-config.h

```diff
@@ -70,6 +70,7 @@ struct link_config {
         size_t speed;
         Duplex duplex;
         int autonegotiation;
+        uint32_t advertising[ETHTOOL_LINK_MODE_MASK_MAX_KERNEL_NU32];
```

Comment:
hmm, not sure I grok this... you define this as uint32_t array here, but the option actually only fills in one value, and that of type NetDevAdvertise, which isn't uin32_t?

Why is this an array if we never fill in more than one value? Why don't the types match?

Reply:
This is exactly how kernel defines . Need to retain the same. http://elixir.free-electrons.com/linux/latest/source/net/core/ethtool.c#L535

http://elixir.free-electrons.com/linux/latest/source/net/core/ethtool.c#L680

Reply:
hmm, but why does it do that?

I really don't know the kernel's APIs in this case, but is it possible that instead of configuring a single value you actually configure a set of supported modes to advertise, and thus the field is an array, so that you can define multiple?

Reply:
No I don't think it's  possible since the kernel code shows that copy_to_user or copy_from_user copies the exact size.

Reply:
And what does it do with the rest of the data?

Reply:
Looked at the driver code of bnx . seems not used only one uint32 type is used.

Here is the commit though. https://github.com/torvalds/linux/commit/3f1ac7a700d039c61d8d8b99f28d605d489a60cf

Reply:
The kernel is prepared to allow higher flags. Right now only bits up to 30 are used, so there is no need. I guess it's OK if we do the same, even though now only the first element of the array is used.